### PR TITLE
[MB-1727] - Subscriptions don't become inactive once the db connection is dropped and restored

### PR DESCRIPTION
**Pull request by sdkottegoda**
Addresses the issue reported at https://wso2.org/jira/browse/MB-1727

### CODE REVIEW

**Comment by ramith** [comment]
Could we attach a jira ID and put a descriptive name of the PR.
and make it a one commit? @sdkottegoda

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/FlowControlManager.java [hunk 1]
```diff
@@ -376,6 +376,7 @@ public synchronized void prepareChannelsForShutdown () {
     @Override
     public void storeNonOperational(HealthAwareStore store, Exception ex) {
         //Remove all local subscriptions and enable error based flow control when the store becomes non-operational.
+        System.out.println("Flow control manager: store non operational.");
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/FlowControlManager.java]: why are we doing a SysOUT ?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/ClusterManager.java [hunk 2]
```diff
@@ -314,6 +314,7 @@ public boolean getStoreHealth() {
      */
     @Override
     public void storeNonOperational(HealthAwareStore store, Exception ex) {
+        System.out.println("Cluster manager: store non operational.");
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/ClusterManager.java]: hope this sysOUT will not get merged.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingAndesContextStore.java [hunk 3]
```diff
@@ -59,12 +67,16 @@
      */
     private ScheduledFuture<?> storeHealthDetectingFuture;
 
+    private AtomicBoolean isStoreAvailable;
```
Comment (ramith) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/store/FailureObservingAndesContextStore.java]: we will document this right?

**Comment by pumudu88** [comment]
Can we add some description on fixed issues?

**Comment by a5anka** [comment]
It is better to have a meaningful commit message

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesRecoveryTask.java [hunk 4]
```diff
@@ -138,7 +138,12 @@ public void storeNonOperational(HealthAwareStore store, Exception ex) {
 	public void storeOperational(HealthAwareStore store) {
 		if (store.getClass().getSuperclass().isInstance(AndesContextStore.class)) {
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesRecoveryTask.java]: Why would we need this check?

**Pull request merged**